Issue 42420: Support source query filter via ETL XML

labkey-jeckels opened the pull request:
#### Rationale
To make it easier to create ETLs that run against a remote server where the ETL developer may not have permission to create custom queries, it's more convenient to allow the ETL definition to add filters to the source table/query.

#### Changes
* Add filters to ETL XSD
* Parse and propagate them
* Apply them when fetching the source data

Commit:
Issue 42420: Support source query filter via ETL XML

Commit:
Introduce Jackson deserializer so there's no need for a separate copy of the Filter class

Update expected number of transforms to fix test

bbimber commented:
@labkey-jeckels So does this count as clean test results? if you have other thoughts on the Filter serialization problem I'm happy to try to implement something other than what I proposed here: https://github.com/LabKey/labkey-api-java/pull/17/files

labkey-jeckels commented:
Yes, the tests are looking happy but as Marty mentioned here:

https://github.com/LabKey/dataintegration/pull/90

the same filter should be applied to the check for modified data query. I'm mid-refactor to do that.

bbimber commented:
@labkey-jeckels OK, that is a good point. If sound like you're already doing that change, but feel free to push any of this back to me if it would help get this closed out.

Commit:
Respect filters during the check for updated rows

Add test coverage for filtered update checks

Eliminate overloaded methods variants
